modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/util/Commons.java

```diff
@@ -861,4 +864,46 @@ public static SqlQueryType getQueryType(SqlNode sqlNode) {
                 return null;
         }
     }
+
+    /** Returns the minimum unscaled value of a numeric type.
+     *
+     * @param type a numeric type
+     */
+    public static long getMinValue(RelDataType type) {
+        SqlTypeName typeName = type.getSqlTypeName();
+        switch (typeName) {
+            case TINYINT:
+                return Byte.MIN_VALUE;
+            case SMALLINT:
+                return Short.MIN_VALUE;
+            case INTEGER:
+                return Integer.MIN_VALUE;
+            case BIGINT:
+            case DECIMAL:
+                return NumberUtil.getMinUnscaled(type.getPrecision()).longValue();
+            default:
+                throw new AssertionError("getMinValue(" + typeName + ")");
+        }
+    }
+
+    /** Returns the maximum unscaled value of a numeric type.
+     *
+     * @param type a numeric type
+     */
+    public static long getMaxValue(RelDataType type) {
```

Comment:
Both this and `getMinValue` can be removed in favour of calcite's `SqlTypeName::getLimit`